src/filters/dasher.c

```diff
@@ -4163,6 +4163,23 @@ static void dasher_setup_sources(GF_Filter *filter, GF_DasherCtx *ctx, GF_MPD_Ad
 			}
 		}
 
+		//update startNumber based on first cts
+		if (ctx->segcts) {
+			GF_FilterPacket *pck = gf_filter_pid_get_packet(ds->ipid);
+			if (pck) {
+				u64 seg_dur = (u64) (ds->dash_dur.num) * ds->timescale / ds->dash_dur.den;
+				u64 cts = gf_filter_pck_get_cts(pck);
+				if (ds->stream_type == GF_STREAM_AUDIO)
+					cts += ds->presentation_time_offset;
+				if (ds->timescale != ds->mpd_timescale)
+					cts = gf_timestamp_rescale(cts, ds->timescale, ds->mpd_timescale);
+				ds->startNumber = (u32) (cts / seg_dur) + 1;
```

Comment:
shouldn't we also take into account user-defined start number GF_PROP_PID_START_NUMBER ? If set, add it.